### ISSUE TITLE
Support version bumps when version is part of the executable name

### ISSUE THREAD

**Issue by corneliusweig** [ahmetb/krew-index-autoapprove#2]
When the executable in a manifest contains the version, the version bump is currently not recognized as a version bump.

To support this, the tool should

- check if a hunk includes a change to the executable
- check if the executable in the old manifest contains the old version
  - if so, ensure that the executable in the new manifest contains the new version
- otherwise, fail the auto-merge (better safe than sorry)

For example, https://github.com/kubernetes-sigs/krew-index/pull/234

**Comment by corneliusweig** [comment]
> We should discourage people from making version part of the executable name.
>
> It makes it harder not only for Krew but also for manual installation. It's already in a version-branded .tar.gz/.zip file, why add version number to the executable as well?

From https://github.com/ahmetb/krew-index-autoapprove/issues/1#issuecomment-530146584

**Comment by corneliusweig** [comment]
Yes, that's even cleaner. Not everyone will follow this recommendation though. And that means more manual review work, unless this is implemented (despite being discouraged).

**Comment by ahmetb** [comment]
I currently don't see any such plugins in the index, also many of them use globbing with `**` to avoid listing version strings in `from:` spec.
